### PR TITLE
fix bug to support dropout eval grad computing.

### PR DESCRIPTION
**Pull request by limin2021**
### PR types
Bug fixes

### PR changes
OPs

### Describe 
Question: 
#35122 support dropout in eval mode (paddle2.0已经将eval和no_grad解绑，eval下做反向是合理的，竞品也都支持这个行为). But #35621 remove these modifications of #35122. 
In this PR, we recover the modifications in #35122. 

test code:
image

before this PR:
image

after this PR:
image

### CODE REVIEW

**Review by wangxicoding — APPROVED** [review]
LGTM

**Review by smallv0221 — APPROVED** [review]
LGTM

File: paddle/fluid/operators/dropout_impl.cu.h [hunk 1]
```diff
@@ -244,34 +244,42 @@ void DropoutGradGPUKernelDriver(const platform::CUDADeviceContext& dev_ctx,
...
   auto dX = EigenVector<T>::Flatten(*grad_x);
   auto dY = EigenVector<T>::Flatten(grad_y);
 
   auto& place = *dev_ctx.eigen_device();
-  if (dropout_implementation == "upscale_in_train") {
-    if (dropout_prob == 1.0f) {
-      dX.device(place) = static_cast<T>(0) * dY;
+  if (is_test) {
+    if (dropout_implementation == "upscale_in_train") {
+      dX.device(place) = static_cast<T>(1) * dY;
```
Comment (wangxicoding) [inline paddle/fluid/operators/dropout_impl.cu.h:254]: 是不是memcpy好一些

Comment (limin2021) [reply paddle/fluid/operators/dropout_impl.cu.h:254]: 嗯嗯，该pr只用于恢复 #35122 的修改哈。后续可以进一步优化。

**Pull request merged**